### ISSUE TITLE
Add Half support to Microsoft.Data.Analysis

### ISSUE THREAD

**Issue by eerhardt** [dotnet/corefxlab#2930]
With https://github.com/dotnet/runtime/issues/936 and https://github.com/dotnet/corefxlab/pull/2929, .NET is getting a `System.Half` type. We should support this type in `DataFrame`, just like all the other built-in value types.

cc @pgovind

**Comment by kevin-wittmer** [comment]
@eerhardt I would like to work on this issue. Can you assign it to me please?

**Comment by pgovind** [comment]
@kevin-wittmer : Let us know if you have questions. We're happy to help!

**Comment by kevin-wittmer** [comment]
@pgovind `Half` should be convertible to `decimal`, right? I'm just looking at `Half.v1.cs` and it looks like there isn't an operator defined for the conversion, which is currently causing me problems.

**Comment by pgovind** [comment]
>  Half should be convertible to decimal, right?

You could cast it to `float` or `double` and then to `decimal`?

**Comment by kevin-wittmer** [comment]
@pgovind Sure, but that doesn't work well with how `Coverters.tt` works. That template generates a class that converts directly from `Half` to `decimal`.

**Comment by pgovind** [comment]
I see. I'm assuming you modified `primitiveTypeToPrimitivityLevelMap` and inserted `Half` in there somewhere? I have a couple ideas. Can you paste the code that is causing problems here?

**Comment by kevin-wittmer** [comment]
Yeah so I added `{"Half", 4}` to `primitiveTypeToPrimitivityLevelMap` like so:
```
public Dictionary<string, int> primitiveTypeToPrimitivityLevelMap = new Dictionary<string, int> {
	{"byte", 1},
	{"sbyte", 1},
	{"short", 2},
	{"ushort", 2},
	{"int", 3},
	{"uint", 3},
	{"long", 4},
	{"ulong", 4},
	{"Half", 4},
	{"float", 5},
	{"double", 6},
	{"decimal", 7}
};
```
Which generates the following converter:
```
internal class HalfDecimalConverter : IDecimalConverter<Half>
{
	public decimal GetDecimal(Half value)
	{
		return (decimal)value;
	}
}
```
But the compiler isn't happy because the operator for casting `Half` to `decimal` isn't defined.

**Comment by pgovind** [comment]
Does this work?
```
    internal class <#=capitalizedLoopPrimitive#><#=typeFirstCharUpper#>Converter : I<#=typeFirstCharUpper#>Converter<<#=typeLoop.Key#>>
    {
        public <#=type#> Get<#=typeFirstCharUpper#>(<#=typeLoop.Key#> value)
        {
<# 
        if (typeLoop.Key == "Half")
        {
#>
            return (<#=type#>)(float)value;
<# 
        }
        else
        {
#>
            return (<#=type#>)value;
<# 
        }
#>
        }
    }
```

Basically I'm special casing whenever we see a `Half` to add an extra cast to `(float)`
